tasks/basecalling/task_samtools_convert.wdl

```diff
@@ -0,0 +1,42 @@
+version 1.0
+
+task samtools_convert {
+  input {
+    Array[File] sam_files
+    Int cpu = 4
+    Int memory = 16
+    Int disk_size = 50
+    String docker = "us-docker.pkg.dev/general-theiagen/staphb/samtools:1.15"
+  }
+
+  command <<< 
+    set -e
+
+    bam_output="output/bam/"
+    mkdir -p $bam_output
+
+    for sam_file in ~{sep=" " sam_files}; do
+      base_name=$(basename $sam_file .sam)
+      bam_file="${bam_output}/${base_name}.bam"
+
+      echo "Converting SAM to BAM: $sam_file -> $bam_file"
+      samtools view -bS "$sam_file" > "$bam_file" || { echo "ERROR: samtools failed to convert $sam_file to BAM" >&2; exit 1; }
```

Comment:
not sure if samtools automatically uses all cpus available so I would add the option to pass in number of cpus: `samtools view -@ $(nproc) ....`